.github/workflows/build.yml

```diff
@@ -0,0 +1,33 @@
+name: Build
+on:
+  pull_request:
+
+env:
+  SONAR_ORG: flexbase
+  SONAR_PROJECT_KEY: flexbase-eng_http-client-middleware
+  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+  SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        fetch-depth: 0 # Disabling shallow clone is recommended for improving relevancy of reporting
+    - uses: actions/setup-node@v3
+    - name: yarn install
+      run: yarn install
+    - name: Build
```

Comment:
If you want, break this up into several jobs - they will run in parallel.

then the last job, sonarcloud scanning can run with a needs statement for the final job like so: needs: [build, lint, test] to ensure the first 3 parallel jobs are green before scanning.

sonarcloud i think needs the code coverage file to upload to the cloud for it to be able to report out diffs and set the webhook comment in PR

cc @br-carter

Reply:
Tried that and the checkout action was cleaning up before sonar ran so the coverage file was lost. Personally would prefer it to work the way you described but I was getting fed up with it and just wanted to get it up...